rust/lance/src/dataset/write/merge_insert.rs

```diff
@@ -3138,4 +3241,405 @@ mod tests {
             StreamingTableExec: partition_sizes=1, projection=[value, key]"
         ).await.unwrap();
     }
+
+    #[tokio::test]
+    async fn test_explain_plan() {
```

Comment:
It seems like we should be able to consolidate these test cases. For example, the  `test_explain_plan_string_output` could be part of the basic `test_explain_plan`.